public/app/features/alerting/unified/components/receivers/form/GrafanaReceiverForm.tsx

```diff
@@ -81,16 +86,23 @@ export const GrafanaReceiverForm = ({ contactPoint, readOnly = false, editMode }
 
   const onSubmit = async (values: ReceiverFormValues<GrafanaChannelValues>) => {
     const newReceiver = formValuesToGrafanaReceiver(values, id2original, defaultChannelValues, grafanaNotifiers);
+
     try {
       if (editMode) {
-        await updateContactPoint({
-          contactPoint: newReceiver,
-          id: contactPoint!.id,
-          resourceVersion: contactPoint?.metadata?.resourceVersion,
-          originalName: contactPoint?.name,
-        });
+        if (useK8sAPI && contactPoint && contactPoint.id) {
+          await updateContactPoint.execute({
+            contactPoint: newReceiver,
+            id: contactPoint.id,
+            resourceVersion: contactPoint?.metadata?.resourceVersion,
+          });
+        } else if (contactPoint) {
+          await updateContactPoint.execute({
+            contactPoint: newReceiver,
+            originalName: contactPoint.name,
+          });
+        }
       } else {
-        await createContactPoint({ contactPoint: newReceiver });
+        await createContactPoint.execute({ contactPoint: newReceiver });
       }
       locationService.push('/alerting/notifications');
     } catch (error) {
```

Comment:
I wasn't too sure about this bit, as this was catching (and not logging) exceptions thrown by the reducer. 
@tomratcliffe any thoughts on how to approach error handling here? This seems to swallow errors and not log them or show them in the form 🤔